### PR TITLE
Turn on validate customers and hmac by default

### PR DESCRIPTION
**Pull request by martindekov**
Setting the customer validation and hmac validation on by
default also moving customer validation into its own function
so it can be tested more easily and extending the tests
to showcase the logic the work was started by @ivanayov

Signed-off-by: Martin Dekov (VMware) <mdekov@vmware.com>

## Description

This adds more secure experience for the user so the hmac and security validation should now be turned off intentionally with explicit `false` or `0`

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md))

Actually part of [#316](https://github.com/openfaas/openfaas-cloud/issues/316) issue in OpenFaaS Cloud

## How Has This Been Tested?

Unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

For derek users with authentication turned off if they move to the newest version, they should explicitly set the `validate_customers: 0/false` and `validate_hmac:0/false`

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: auth/customers.go [hunk 1]
```diff
@@ -32,10 +32,8 @@ func buildCustomerURL() string {
 // IsCustomer returns true if a customer is listed in the customers file.
 // The validation is controlled by the 'validate_customers' env-var
 func IsCustomer(ownerLogin string, c *http.Client) (bool, error) {
-	validate := os.Getenv("validate_customers")
-
-	if len(validate) == 0 || (validate == "false" || validate == "0") {
-
+	validate := customerValidation()
```
Comment (alexellis) [inline auth/customers.go]: customerValidation could be `useCustomerValidation` or `customerValidationEnabled` to show the intent. It sounds a bike like an action as it is.

Comment (martindekov) [reply auth/customers.go]: Good idea, will quickly amend it

File: main_test.go [hunk 2]
```diff
@@ -37,12 +37,40 @@ func Test_getContributingURL(t *testing.T) {
...
-		t.Errorf("want %t got %t", want, enabled)
-		t.Fail()
+func Test_customerValidation(t *testing.T) {
+	tests := []struct {
+		title        string
+		value        string
+		expectedBool bool
+	}{
+		{
+			title:        "`validate_hmac` is unset",
```
Comment (alexellis) [inline main_test.go]: How about? ``validate_hmac` is unset, defaults to on`

File: auth/customers_test.go [hunk 3]
```diff
@@ -47,3 +47,41 @@ func Test_findCustomerURL(t *testing.T) {
 		})
 	}
 }
+
+func Test_customerValidationEnabled(t *testing.T) {
```
Comment (alexellis) [inline auth/customers_test.go]: 👏 💯 😄

File: main_test.go (hunk 2 above)
Comment (martindekov) [reply main_test.go]: Sounds good ^

**Pull request merged**